controllers/authpolicy_authconfig.go

```diff
@@ -187,8 +194,10 @@ func authorinoSpecsFromConfigs[T, U any](configs map[string]U, extractAuthorinoS
 }
 
 func mergeConditionsFromRouteSelectorsIntoConfigs(ap *api.AuthPolicy, route *gatewayapiv1.HTTPRoute, authConfig *authorinoapi.AuthConfig) (*authorinoapi.AuthConfig, error) {
+	commonSpec := ap.Spec.CommonSpec()
+
```

Comment:
There's no `nil` check for `AuthScheme` here as this function is only called from `r.desiredAuthConfig()` which has been updated with this `nil` check for `AuthScheme`. This function will not be called in this case